### PR TITLE
[node] Add missing console constructor variants

### LINKED ISSUE #30339: Node.js 10.11 'Console' definition does not meet spec
- [x] I tried using the `@types/node` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @DefinitelyTyped, @tdreyno jkomyno, @a-tarasyuk, @alvis, @r3nya, @brunoscheufler, @smac89, @tellnes, @DeividasBakanas, @eyqs, @Flarna, @Hannes-Magnusson-CK, @KSXGitHub, @hoo29, @kjin, @ajafff, @islishude, @mwiktorczyk, @matthieusieben, @mohsen1, @n-e, @octo-sniffle, @parambirs, @eps1lon, @SimonSchick, @ThomasdenH, @WilcoBakker, @wwwy3y3, @ZaneHannanAU

`// Type definitions for Node.js 10.11`
`// Project: http://nodejs.org/`

After typing all those author tags, I hope I'm not wrong in thinking that the Node `Console` class ought to have type defintions two forms of constructor, as described on NodeJS.org site[1]:

`Version | Changes
-- | --
v10.0.0 | The Console constructor now supports an options argument, and the colorMode option was introduced.

[1]: https://nodejs.org/api/console.html#console_new_console_stdout_stderr_ignoreerrors

### PR DESCRIPTION
**Pull request by Flarna**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/dist/latest-v11.x/docs/api/console.html#console_new_console_stdout_stderr_ignoreerrors
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Fixes #30339

### CODE REVIEW

**Review by ZaneHannanAU — APPROVED** [review]

**Comment by ZaneHannanAU** [comment]
Wait holy shit the build passed?
Did we deprecate old ones or not yet?

**Review by islishude — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]

**Comment by Flarna** [comment]
@andy-ms @RyanCavanaugh would be nice if this could be merged as test fail seems to be because of #30323

File: types/node/index.d.ts [hunk 1]
```diff
@@ -497,9 +497,17 @@ declare namespace NodeJS {
         sorted?: boolean | ((a: string, b: string) => number);
     }
 
+    interface ConsoleConstructorOptions {
+        stdout: WritableStream,
```
Comment (andy-ms) [inline types/node/index.d.ts]: There are lint errors here -- should use `;` not `,`

Comment (Flarna) [reply types/node/index.d.ts]: ups, seems I'm getting too fast with pointing to unrelated fails. should be fixed.

**Pull request merged**